### ISSUE TITLE
Ocalmfuse crash

### ISSUE THREAD

**Issue by kattycris** [astrada/google-drive-ocamlfuse#434]
I am using the ocalmfuse in an 8 cores vps with 8gb of ram, and all the time my drives give the problem, I have to umount and mount the drive to go back. Do you know what it can be?

**Comment by astrada** [comment]
If the process crashes, you should enable debug logging and check the logs (https://github.com/astrada/google-drive-ocamlfuse#troubleshooting). If there is no errors in the logs, it can be that you are using too much memory (have you enabled streaming? have you increased buffer sizes). If you are using plex (or something like that) there are probably access patterns that lock the process. Unfortunately, I cannot reproduce them since I don't have plex, so I don't know how to fix the issue.

**Comment by link0087** [comment]
Hi, 
i have the same problem too, the real problem is that it is not easy to detect when it crash. there is a way to mount all on debug mode without left the shell opened? beacuase is somenthing that happen every day but randomly so i cant left the shell opened on debug mode.


here is my configuration:

```
apps_script_format=json
apps_script_icon=
async_upload=true
cache_directory=
client_id=MY-CLIENT-ID
client_secret=MY-CLIENT-SECRET
connect_timeout_ms=5000
curl_debug_off=false
data_directory=
delete_forever_in_trash_folder=false
docs_file_extension=true
document_format=odt
document_icon=
download_docs=true
drawing_format=png
drawing_icon=
form_format=zip
form_icon=
fusion_table_format=desktop
fusion_table_icon=
keep_duplicates=false
large_file_read_only=false
large_file_threshold_mb=16
log_directory=
lost_and_found=false
low_speed_limit=0
low_speed_time=0
map_format=desktop
map_icon=
max_cache_size_mb=512
max_download_speed=0
max_memory_cache_size=10485760
max_retries=8
max_upload_chunk_size=1099511627776
max_upload_speed=0
memory_buffer_size=1048576
metadata_cache_time=60
presentation_format=pdf
presentation_icon=
read_ahead_buffers=3
read_only=false
root_folder=
shared_with_me=false
spreadsheet_format=ods
spreadsheet_icon=
sqlite3_busy_timeout=20
stream_large_files=true
team_drive_id=
umask=0o1411
verification_code=
```

My server is ubuntu 16.04

Thank you

F.

**Comment by link0087** [comment]
up

**Comment by inventor96** [comment]
@link0087 I've created this cronjob to enable debug logging in the background:
`@reboot sleep 10; google-drive-ocamlfuse -debug -config /opt/gdfuse/default/config /opt/gdrive > /opt/gdfuse/gdfuse.cron.log 2>&1`
And I have this setting in my config file: `log_directory=/opt/gdfuse/default`
Because it's running as a cron job and not a user, I've explicitly specified the other directory values, too.

With all this set up, I get the debug logging in /opt/gdfuse/default/gdfuse.log, and any output from the command itself in /opt/gdfuse/gdfuse.cron.log

**Comment by kristofejro** [comment]
I have very similar problem on my Raspberry Pi.
It started since upgrade to 0.7.5.
When I mount google drive with google-drive-ocamlfuse it starts to eat memory until there is nothing left.
It is eating in intervals ~60 seconds - I was monitoring it with 'top'
Unfortunatelly I couldn't trace where is the problem - no useful information in any log. Cleaning cache of google-drive-ocamlfuse also didn't help.

**Comment by astrada** [comment]
@kristofejro thanks for your feedback. It's a problem with a dependency: a workaround (if you are using opam) is to run:

    opam pin add gapi-ocaml 0.3.10

When it's fixed you can remove the pin with:

    opam pin remove gapi-ocaml

**Comment by kristofejro** [comment]
Thanks @astrada for explanation.
So far I have installed google-drive-ocamlfuse from your ppa (for Debian Stretch) and it works just fine.
I will try to pin gapi-ocaml and then check.

**Comment by astrada** [comment]
Should be fixed in 0.7.6 (with gapi-ocaml 0.3.12). I'm uploading this version to the beta [ppa](https://launchpad.net/~alessandro-strada/+archive/ubuntu/google-drive-ocamlfuse-beta) and opam.

**Comment by kristofejro** [comment]
Pinning gapi-ocaml to version 0.3.10 really helpped.
